### PR TITLE
feat: disable OTC flow for community credits

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1732

Disable OTC flow for community credit for the following pages:
- project details
- credit class

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. Project details: https://deploy-preview-1932--regen-registry.netlify.app/credit-classes/C46
2. Credit class details: https://deploy-preview-1932--regen-registry.netlify.app/project/C46-001 

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by erikalogie** [comment]
> @erikalogie you can check that the OTC is disabled for community credit here:
> 
> 1. Project details: https://deploy-preview-1932--regen-registry.netlify.app/credit-classes/C46
> 2. Credit class details: https://deploy-preview-1932--regen-registry.netlify.app/project/C46-001
> 
> I've temporarily renamed C46 credit class in sanity to make it like a community credit class.

Great looks good. Only question I have is whether we really need the "Choose how to purchase credits" popup as there is only one option now, maybe we can just go straight into the rest of the flow depending on whether the user is logged in or out.

**Comment by flagrede** [comment]
@erikalogie This would involve a lot more changes as we would need to partly revert the old behavior to show the error banner if the user tries to buy while being not connected but still handle the case with options in the modal. If you think it's not a big deal it would save us some time to keep it like that.

**Review by blushi — COMMENTED** [review]
Tried to buy from C04-001-20180202-20190202-001 which should be considered community credits since C04 is not on sanity staging but still got the OTC info card:

image

**Comment by erikalogie** [comment]
> Tried to buy from C04-001-20180202-20190202-001 which should be considered community credits since C04 is not on sanity staging but still got the OTC info card:
> 
> image

Oh yeah I totally missed this during QA, good call @blushi

**Comment by flagrede** [comment]
@blushi @erikalogie thanks for catching that, will investigate.

**Comment by flagrede** [comment]
@blushi @erikalogie this should be fixed, I actually missed this part.

**Review by blushi — APPROVED** [review]
looks good
just noticed a bug but not related to the changes here, this should be fixed in a separate issue, it doesn't happen on prod but I guess it should be fixed nevertheless

File: web-registry/src/pages/Marketplace/Storefront/Storefront.tsx [hunk 1]
```diff
@@ -186,6 +186,8 @@ export const Storefront = (): JSX.Element => {
     disableAutoRetire,
   } = normalizedSellOrders[selectedSellOrder ?? 0] ?? {};
 
+  const isCommunityCredit = project?.classId === project?.classIdOrName;
```
[the remark this answers is not in the document]
Comment (flagrede) [reply web-registry/src/pages/Marketplace/Storefront/Storefront.tsx:189]: Will add the comment. Seems like there's a data issue on staging/redwood for the issue you found out.

**Pull request merged**